GameDesign/Assets/Scripts/MonoBehaviours/Movements/NPCMovementManager.cs

```diff
@@ -18,19 +20,26 @@ void Start()
             agent = (UnityEngine.AI.NavMeshAgent)GetComponent<UnityEngine.AI.NavMeshAgent>();
             if (agent == null) Debug.LogError("Could not find NavMeshAgent");
         }
+        if (targetWayPoint == null)
+        {
+            Debug.LogError($"{this.name}: Missing initial variable targetWaypoint.");
+        }
 
-        agent.SetDestination(previousWayPoint.transform.position);
+        agent.SetDestination(targetWayPoint.transform.position);
 
     }
 
 
     void Update()
     {
-        if (Vector3.Distance(transform.position, targetWayPoint.transform.position) < distanceThreshHoldToReachWP)
+        if (TTL > MaxTimeToReachWaypoint || Vector3.Distance(transform.position, targetWayPoint.transform.position) < distanceThreshHoldToReachWP)
```

Comment:
What happens if we get to TTL and still haven't reached the waypoint but we invoke the "waypointReached"? For now we will just get the next waypoint, but this depends on the implementation of the interface method. For example, if a person doesn't get to the despawner quickly enough, it will disappear mid game?

Reply:
waypoint reached is the interface, the despawner could do a check and handle it itself
the person just tells the despawner to run the function. the despawner could check if its within certain distance and do that else just set the target to ehter itself or other despawner